src/cgalutils-applyops.cc

```diff
@@ -34,8 +34,43 @@
 #include <queue>
 #include <unordered_set>
 
+#include <atomic>
+
 namespace CGALUtils {
 
+	// manage changes to CGAL::set_error_behaviour:
+	//  a mutex to guard access
+	std::atomic_flag lockedErrorsLock = ATOMIC_FLAG_INIT;
+	//  the number of times lockErrors has been called (without corresponding unlockErrors)
+	size_t lockedErrorsCount = 0;
+	//  the original error state
+	CGAL::Failure_behaviour lockedErrors = CGAL::Failure_behaviour::ABORT;
+
+#define SPIN_LOCK(flag) do { } while(flag.test_and_set())
+
+	// increments lockedErrorsCount and calls CGAL::set_error_behavior if this is the first [unlocked] call
+	void lockErrors(CGAL::Failure_behaviour behavior)
```

Comment:
While we're already doing the work to clean up the cgal error handling, could we take it one step further and encapsulate this into a RAII-style lock, and use the appropriate C++11 constructs for that + boost's spinlock? e.g.
```
	class ErrorLocker {
	public:
	  ErrorLocker() {
	    std::lock_guard<boost::detail::spinlock> guard(ErrorLocker::lock);
	    oldBehavior = CGAL::set_error_behaviour(CGAL::THROW_EXCEPTION);
	  }
	  ~ErrorLocker() {
	    std::lock_guard<boost::detail::spinlock> guard(ErrorLocker::lock);
	    CGAL::set_error_behaviour(oldBehavior);
	  }
	private:
	  CGAL::Failure_behaviour oldBehavior;
	  static boost::detail::spinlock lock;
	};
```